sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/models/CosmosOperationDetails.java

```diff
@@ -0,0 +1,58 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+package com.azure.cosmos.models;
+
+import com.azure.cosmos.CosmosDiagnosticsContext;
+import com.azure.cosmos.implementation.ImplementationBridgeHelpers;
+import com.azure.cosmos.implementation.OverridableRequestOptions;
+
+/**
+ * Encapsulates the details of an operation and allows for overriding some properties in the internal request options.
+ *
+ */
+public final class CosmosOperationDetails {
+
+    private final OverridableRequestOptions requestOptions;
+    private final CosmosDiagnosticsContext cosmosDiagnosticsContext;
+
+    CosmosOperationDetails(OverridableRequestOptions requestOptions, CosmosDiagnosticsContext cosmosDiagnosticsContext) {
+        this.requestOptions = requestOptions;
+        this.cosmosDiagnosticsContext = cosmosDiagnosticsContext;
+    }
+
+    /**
+     * Applies the options in the CosmosCommonRequestOptions to the internal request options.
+     *
+     * @param readOnlyRequestOptions the common request options for overriding.
+     */
+    public void setCommonOptions(ReadOnlyRequestOptions readOnlyRequestOptions) {
```

Comment:
also wonder why ReadOnlyRequestOptions, customer can construct this option by themselves, so not just read only?